provider/pkg/overlays/examples/callbackFunction.md

```diff
@@ -0,0 +1,216 @@
+A CallbackFunction is a special type of `aws.lambda.Function` that can be created out of an actual JavaScript function instance. The Pulumi compiler and runtime work in tandem to extract your function, package it up along with its dependencies, upload the package to AWS Lambda, and configure the resulting AWS Lambda resources automatically.
+
+The JavaScript function may capture references to other variables in the surrounding code, including other resources and even imported modules. The Pulumi compiler figures out how to serialize the resulting closure as it uploads and configures the AWS Lambda. This works even if you are composing multiple functions together.
+
+See [Function Serialization](https://www.pulumi.com/docs/concepts/inputs-outputs/function-serialization/) for additional details on this process.
+
+If no IAM Role is specified, CallbackFunction will automatically use the following managed policies:
+- `AWSLambda_FullAccess`
+- `CloudWatchFullAccessV2`
+- `CloudWatchEventsFullAccess`
+- `AmazonS3FullAccess`
+- `AmazonDynamoDBFullAccess`
+- `AmazonSQSFullAccess`
+- `AmazonKinesisFullAccess`
+- `AWSCloudFormationReadOnlyAccess`
+- `AmazonCognitoPowerUser`
+- `AWSXrayWriteOnlyAccess`
+
+### Lambda Function Handler
+
+You can provide the JavaScript function used for the Lambda Function's Handler either directly by setting the `callback` input property or instead specify the `callbackFactory`, which is a Javascript function that will be called to produce the callback function that is the entrypoint for the AWS Lambda.
+Using `callbackFactory` is useful when there is expensive initialization work that should only be executed once. The factory-function will be invoked once when the final AWS Lambda module is loaded. It can run whatever code it needs, and will end by returning the actual function that Lambda will call into each time the Lambda is invoked.
+
+The callback function should follow the following signature: `(event: E, context: Context, callback: (error?: Error | string | null, result?: R) => void) => Promise<R> | void`. It is recommended to use an async function, otherwise the Lambda execution will run until the `callback` parameter is called and the event loop is empty. See [Define Lambda function handler in Node.js](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html) for additional details.
```

Comment:
Consider avoiding inline code gory details in favor of a code example (or just entirely drop).